docs/howto/composite-primary-key.txt

```diff
@@ -0,0 +1,159 @@
+    1
+
+Migrating to a composite primary key
+====================================
+
+Django doesn't support migrating to, or from, a composite primary key after the
+table is created.
+
+If you would like to migrate an existing table from a single primary key to a
+composite primary key, follow your database backend's instructions to do so.
+
+Once the composite primary key is in place, add the ``CompositePrimaryKey``
+field to your model. This allows Django to recognize and handle the composite
+primary key appropriately.
+
+While migration operations (e.g. ``AddField``, ``AlterField``) on
+primary key fields are not supported, ``makemigrations`` will still detect
+changes.
+
+In order to avoid errors, it's recommended to apply such migrations with
+``--fake``.
+
+Alternatively, :class:`.SeparateDatabaseAndState` may be used to execute
+the backend-specific migrations and Django-generated migrations in
+a single operation.
+
+How to define a composite foreign key
+=====================================
+
+At the moment, relationship fields (``ForeignKey``, ``OneToOneField``,
+``ManyToManyField``) don't support composite primary keys.
+
+So, given the ``OrderLineItem`` model, we cannot do:
+
+.. code-block:: pycon
+
+    class Foo(models.Model):
+        item = models.ForeignKey(OrderLineItem, on_delete=models.CASCADE)
+
+Because ``ForeignKey`` cannot reference models with composite primary keys
+*yet*.
+
+To work around this limitation, ``ForeignObject`` can be used as an
+alternative:
+
+.. code-block:: pycon
+
+    class Foo(models.Model):
+        item_order_id = models.IntegerField()
+        item_product_id = models.CharField(max_length=20)
+        item = models.ForeignObject(
+            OrderLineItem,
+            on_delete=models.CASCADE,
+            from_fields=("item_order_id", "item_product_id"),
+            to_fields=("order_id", "product_id"),
+        )
+
+``ForeignObject`` is much like ``ForeignKey``,
+except that it doesn't create any columns (e.g. ``item_id``), foreign key
+constraints or indexes in the database.
+
+.. warning::
+
+    ``ForeignObject`` is an internal API. The work to provide a public API for
+    composite foreign keys is already under way. Use it at your own risk.
+
+Composite primary keys and generic relations
+============================================
+
+The composite primary key feature is still in its infancy, it doesn't
+support generic relations yet.
+
+The work to add composite primary key support to generic relations
+is already under way.
+
+Composite primary keys and database functions
+=============================================
+
+The database functions may not work with composite primary keys.
+
+This is because most database functions accept a single field only.
+
+.. code-block:: sql
+
+    COUNT("order_id")  -- OK
+    COUNT("product_id", "order_id")  -- ERROR
+
+So, unless the database function accepts multiple fields, don't
+use it with a composite primary key directly.
```

Comment:
```suggestion
These database functions cannot be used with composite primary keys, as
composite primary keys represent multiple fields.
```
Not sure "represent" is the best word :thinking: might be able to say "are a proxy to"